### PR TITLE
Blog app - creating data model

### PR DESCRIPTION
**Pull request by MarcoOquendoC**
# [Blog app - creating data model](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project_v1_1.md)

## Hi good reviewer,

hello

## Entity Relationship Diagram
Entity Relationship Diagram

## This Pull Request generates migration files according to the Entity Relationship Diagram above using the following commands:

- rails generate migration CreateUser name:string photo:string bio:text post_counter:integer
- rails generate migration CreatePost users:references title:string text:text comments_counter:integer likes_counter:integer
- rails generate migration CreateComment users:references posts:references text:text
- rails generate migration CreateLike users:references posts:references

image

image

### CODE REVIEW

File: db/migrate/20230119074348_create_comment.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateComment < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.references :users, null: false, foreign_key: true
```
Comment (VICTOR-CHUKWU) [inline db/migrate/20230119074348_create_comment.rb]: - [ ]  according to project requirement kindly reference user with author i.e `t.references :author, foreign_key: { to_table: 'users' } `

File: db/migrate/20230119074426_create_like.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateLike < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :users, null: false, foreign_key: true
```
Comment (VICTOR-CHUKWU) [inline db/migrate/20230119074426_create_like.rb]: - [ ]  according to project requirement kindly reference user with author i.e `t.references :author, foreign_key: { to_table: 'users' } `

File: db/migrate/20230119073650_create_post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreatePost < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.references :users, null: false, foreign_key: true
```
Comment (VICTOR-CHUKWU) [inline db/migrate/20230119073650_create_post.rb]: - [ ]  according to project requirement kindly reference user with author i.e `t.references :author, foreign_key: { to_table: 'users' } `

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @MarcoOquendoC  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @MarcoOquendoC        :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**